### PR TITLE
Add proper support for Brocade NOS os

### PR DESCRIPTION
**Pull request by vizay**
Added proper support for devices running Brocades NOS os. It should now discover and poll all the standard stuff in Brocades VDX line (cpu, memory, ports etc).

There was also a flag in the nos definition that enabled the functionality to add the ifDescr as ifAlias. This was a legacy fix from Observium, at that time the nos os didn't report ifAlias in a correct way, hence why this was needed. Today however that's not an issue anymore, so that's been cleaned up.

Since Brocade have done something quite funky with their port indexes (ifIndex gives a very long number, instead of the expected iteration from 1 and upwards for each port listed) I've also disabled the printing of the index on the ports page if os == nos.

An alternative way to solve this last thing mentioned could be to add a configuration flag that enables/disables the printing of the ifIndex all in all. Most network vendors of today have pretty straight forward naming conventions for their interfaces, thus making it kind of moot to have the ifIndex printed aswell. That should however probably be discussed in a separate issue :)

### CODE REVIEW

**Comment by vizay** [comment]
Just noticed the conflicts, working on resolving them

**Comment by laf** [comment]
@vizay Thanks. Before we look at the code itself can you resolve the scrutinizer issue (php syntax I believe):

https://scrutinizer-ci.com/g/librenms/librenms/inspections/4d30b2d6-3e8a-4056-bda0-a479b69e42fc/issues/files/html/includes/print-interface.inc.php?status=new&orderField=path&order=asc

**Comment by vizay** [comment]
Fixed :)

had it removed in a few commits before but forgot to remove it fixing the merge.

File: includes/discovery/os/nos.inc.php [hunk 1]
```diff
@@ -1,7 +1,5 @@
 <?php
 
 if (!$os) {
-    if (strstr($sysObjectId, '.1.3.6.1.4.1.1588.2.2.1.1.1.5') || stristr($sysDescr, 'Brocade Communications Systems')) {
-        $os = 'nos';
-    }
-}
+  if (strstr($sysDescr, "Brocade VDX")) { $os = "nos"; }
```
Comment (laf) [inline includes/discovery/os/nos.inc.php]: Will this new check pick up the two checks it's removing or should it be in addition to?

Can you separate the { $os = "nos"; } out as before so we can try and keep the code standards up :)

Comment (vizay) [inline includes/discovery/os/nos.inc.php]: As far as I could find in the MIB documentations, that oid doesn't even exist anymore. checking for Brocade VDX as sysDescr is the correct way and will catch nos devices :)

**Pull request closed without merging**